docs/caduceus.yaml

```diff
@@ -0,0 +1,235 @@
+      summary: Register webhook
+      description:
+        Create a webhook for receiving events.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Webhook'
+        required: true
+      responses:
+        200:
+          description: Webhook registered.
+          content: { }
+        400:
+          $ref: '#/components/responses/400'
+        401:
+          $ref: '#/components/responses/401'
+        403:
+          $ref: '#/components/responses/403'
+        429:
+          $ref: '#/components/responses/429'
+        500:
+          $ref: '#/components/responses/500'
+
+  
+components:
+  schemas:
+    Event:
+      description: Event is the extension of wrp message.
+      externalDocs:
+        url: https://xmidt.io/docs/wrp/simple-messages/#simple-event-definition
+      type: object
+      required:
+       - msg_type
+       - source
+       - dest
+       - payload
+       - sessionID
+      properties:
+        msg_type:
+          type: integer
+          description: The message type for the simple event. This value SHALL be 4.
+          example: 4
+        source:
+          type: string
+          description: The creator of the event.
+          example: "dns:talaria.xmidt.example.com"
+        dest:
+          type: string
+          description: The name of the target device of the event.
+          example: "event:device-status/mac:ffffffffdae4/online"
+        content_type:
+          type: string
+          description: The media type of the payload.
+          example: "application/json"
+        partner_ids:
+          type: array
+          items:
+            type: string
+          description: The list of partner ids the message is meant to target.
+          example: ["example_partner", ".*"]
+        headers:
+          type: array
+          items:
+            type: string
+          description: The headers associated with the payload.
+        metadata:
+          type: object  # map[string]string
+          additionalProperties:
+            type: string
+          description: 
+            The map of name/value pairs used by consumers of WRP messages for 
+            filtering & other purposes.
+          example: {"/boot-time": "1542834188","/last-reconnect-reason": "spanish inquisition" }
+        payload:
+          type: string
+          format: binary
+          description: The bin format packed data.
+        session_id:
+          type: string
+          description: A unique ID for the device's connection session with XMiDT.
+          example: "1aaIkcz1cak0wXzIHsXkV8Uak1V"
+    Webhook:
+      description: The information needed to register a webhook with XMiDT.
+      type: object
+      required:
+      - events
+      properties:
+        registered_from_address:
+          type: string
+          description: The subscription request origin HTTP Address.
+          example: http://127.0.0.1/fakeurl/registerer
```

Comment:
Change to use `https://listener.example.com/registerer` since localhost doesn't work.